### PR TITLE
Feature/basic integration

### PR DESCRIPTION
**Pull request by DC2-DanielKrueger**
**Motivation**

Resolves #<issue>

**Changes**

### CODE REVIEW

File: hivemq-edge/src/main/java/com/hivemq/edge/modules/ModuleLoader.java [hunk 1]
```diff
@@ -133,8 +133,10 @@ protected void loadFromModulesDirectory(final ClassLoader parentClassloader) {
         final ArrayList<Class<? extends T>> classes = new ArrayList<>();
         for (EdgeModule module : modules) {
             try {
+                log.info("Trying to load class '{}' from file '{}'.", serviceClazz, module.root.getAbsoluteFile());
```
Comment (schaebo) [inline hivemq-edge/src/main/java/com/hivemq/edge/modules/ModuleLoader.java:136]: I think these should be trace/debug instead

File: hivemq-edge/src/main/java/com/hivemq/edge/modules/ModuleLoader.java [hunk 2]
```diff
@@ -133,8 +133,10 @@ protected void loadFromModulesDirectory(final ClassLoader parentClassloader) {
...
                 final Iterable<Class<? extends T>> loaded = classServiceLoader.load(serviceClazz, module.classloader);
                 for (Class<? extends T> foundClass : loaded) {
+                    log.info("Adding class '{}' from file '{}'.", foundClass, module.root.getAbsoluteFile());
```
Comment (schaebo) [inline hivemq-edge/src/main/java/com/hivemq/edge/modules/ModuleLoader.java:139]: I think these should be trace/debug instead

File: hivemq-edge/src/main/java/com/hivemq/extensions/core/RestComponentsService.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.hivemq.extensions.core;
+
+import com.hivemq.api.resources.GenericAPIHolder;
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+
+public class RestComponentsService {
```
Comment (schaebo) [inline hivemq-edge/src/main/java/com/hivemq/extensions/core/RestComponentsService.java:6]: may make sense to add an interface here as it is exposed to the modules

Comment (DC2-DanielKrueger) [reply hivemq-edge/src/main/java/com/hivemq/extensions/core/RestComponentsService.java]: yeah ill do so

**Review by schaebo — APPROVED** [review]

**Pull request merged**